catalog/solr/catalog-solr-offline-gazetteer/src/main/resources/OSGI-INF/blueprint/blueprint.xml

```diff
@@ -0,0 +1,39 @@
+<?xml version="1.0" encoding="UTF-8"?>
+<!-- /**
+ * Copyright (c) Codice Foundation
+ *
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser General Public License as published by the Free Software Foundation, either
+ * version 3 of the License, or any later version.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ *
+ **/ -->
+<blueprint xmlns="http://www.osgi.org/xmlns/blueprint/v1.0.0">
+
+  <reference id="solrFactory" interface="org.codice.solr.factory.SolrClientFactory"/>
+
+
+  <bean id="offlineGazetteerPlugin"
+    class="ddf.catalog.solr.offlinegazetteer.OfflineGazetteerPlugin">
+    <argument ref="solrFactory"/>
+  </bean>
+  <service ref="offlineGazetteerPlugin">
+    <interfaces>
+      <value>ddf.catalog.plugin.PostIngestPlugin</value>
+      <value>ddf.catalog.plugin.PreQueryPlugin</value>
+    </interfaces>
+  </service>
+
+  <bean id="executorService" class="java.util.concurrent.Executors"
+    factory-method="newSingleThreadExecutor">
+  </bean>
```

Comment:
❗ I don't expect it to get fixed in this PR but I wanted to say something. Something that really bothers me is our systemic use of separate, single-threaded executors across the entire system. 
- The biggest drawback is likely increased startup time since bootstrapping threads aren't cheap. 
- It makes instrumenting all the separate utility threads a real pain especially if you want a cohesive view of what your system is doing. 
- It may be trivial but most of the time the executor just lingers around when it's done, and while the cost is minimal it would be beneficial and in keeping with a clean runtime to shutdown resources we no longer use or need. 

Some resources: 
- [SO discussion](https://stackoverflow.com/questions/7166915/executors-use-and-cost-penalty)
- [Javadoc](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Executors.html#newSingleThreadExecutor--) but note just from the descriptions that there are more effective options for what we're doing, for example the [cached thread pool](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Executors.html#newCachedThreadPool--) option, which consumes zero resources when no activity is occurring. 
- I think even a [fixed thread pool](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/Executors.html#newFixedThreadPool-int-) initialized early with `~2` threads would be sufficient for all our different "one off executor" operations but that can be allowed to vary on a case-by-case basis.

Reply:
I've moved to a cachedThreadExecutor 👍